assets/js/src/core/modules/data-object/actions/paste/use-paste.tsx

```diff
@@ -0,0 +1,125 @@
+
+interface UsePasteHookParams {
+  storedNode: UseCopyPasteHookReturn['storedNode']
+  nodeTask: UseCopyPasteHookReturn['nodeTask']
+}
+
+export interface UsePasteHookReturn {
+  pasteAsChildTreeContextMenuItem: (node: TreeNodeProps) => ItemType
+  pasteAsChildRecursiveTreeContextMenuItem: (node: TreeNodeProps) => ItemType
+  pasteRecursiveUpdatingReferencesTreeContextMenuItem: (node: TreeNodeProps) => ItemType
+  pasteOnlyContentsTreeContextMenuItem: (node: TreeNodeProps) => ItemType
+  isPasteMenuHidden: (node: Element | TreeNodeProps) => boolean
+}
+
+export const usePaste = ({ storedNode, nodeTask }: UsePasteHookParams): UsePasteHookReturn => {
+  const { t } = useTranslation()
+  const { isTreeActionAllowed } = useTreePermission()
+  const dispatch = useAppDispatch()
+  const { paste } = useCopyPaste('data-object')
+  const { treeId } = useTreeId(true)
+
+  const pasteAsChildRecursiveTreeContextMenuItem = (node: TreeNodeProps): ItemType => {
+    return {
+      label: t('element.tree.paste-as-child-recursive'),
+      key: 'pasteAsChildRecursive',
+      icon: <Icon value={ 'paste' } />,
+      hidden: isPasteOpenHidden(node),
+      onClick: async () => {
+        dispatch(setNodeFetching({ treeId, nodeId: String(node.id), isFetching: true }))
+        await paste(parseInt(node.id), { recursive: true, updateReferences: false }, storedNode)
+      }
+    }
+  }
+
+  const pasteRecursiveUpdatingReferencesTreeContextMenuItem = (node: TreeNodeProps): ItemType => {
+    return {
+      label: t('element.tree.paste-recursive-updating-references'),
+      key: 'pasteRecursiveUpdatingReferences',
+      icon: <Icon value={ 'paste' } />,
+      hidden: isPasteOpenHidden(node),
+      onClick: async () => {
+        dispatch(setNodeFetching({ treeId, nodeId: String(node.id), isFetching: true }))
+        await paste(parseInt(node.id))
+      }
+    }
+  }
+
+  const pasteAsChildTreeContextMenuItem = (node: TreeNodeProps): ItemType => {
+    return {
+      label: t('element.tree.paste-as-child'),
+      key: 'pasteAsChild',
+      icon: <Icon value={ 'paste' } />,
+      hidden: isPasteOpenHidden(node),
+      onClick: async () => {
+        dispatch(setNodeFetching({ treeId, nodeId: String(node.id), isFetching: true }))
+        await paste(parseInt(node.id), { recursive: false, updateReferences: false })
+      }
+    }
+  }
+
+  const pasteOnlyContentsTreeContextMenuItem = (node: TreeNodeProps): ItemType => {
+    return {
+      label: t('element.tree.paste-only-contents'),
+      key: 'pasteOnlyContents',
+      icon: <Icon value={ 'paste' } />,
+      hidden: isPasteOnlyContentsHidden(node),
+      onClick: async () => {
+        console.log('not implemented!')
+      }
+    }
+  }
+
+  const isPasteOpenHidden = (node: Element | TreeNodeProps): boolean => {
+    return !isTreeActionAllowed(TreePermission.Paste) ||
+      (storedNode === undefined || nodeTask !== 'copy') ||
+      !checkElementPermission(node.permissions, 'create')
```

Comment:
Do we need to check the locked status too?

Reply:
nope at least in the old UI its not respected ...

![image](https://github.com/user-attachments/assets/c9392edf-4551-4bf4-b5f9-db1159b2802f)